src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/TstInfo.cs

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using NuGet.Packaging.Signing.DerEncoding;
+
+namespace NuGet.Packaging.Signing
+{
+    /*
+        From RFC 3161 (https://tools.ietf.org/html/rfc3161#section-2.4.2):
+
+            TSTInfo ::= SEQUENCE  {
+               version                      INTEGER  { v1(1) },
+               policy                       TSAPolicyId,
+               messageImprint               MessageImprint,
+                 -- MUST have the same value as the similar field in
+                 -- TimeStampReq
+               serialNumber                 INTEGER,
+                -- Time-Stamping users MUST be ready to accommodate integers
+                -- up to 160 bits.
+               genTime                      GeneralizedTime,
+               accuracy                     Accuracy                 OPTIONAL,
+               ordering                     BOOLEAN             DEFAULT FALSE,
+               nonce                        INTEGER                  OPTIONAL,
+                 -- MUST be present if the similar field was present
+                 -- in TimeStampReq.  In that case it MUST have the same value.
+               tsa                          [0] GeneralName          OPTIONAL,
+               extensions                   [1] IMPLICIT Extensions   OPTIONAL  }
+
+            TSAPolicyId ::= OBJECT IDENTIFIER
+    */
+    /// <remarks>This is public only to facilitate testing.</remarks>
+    public sealed class TstInfo
```

Comment:
there is already a `Rfc3161TimestampTokenInfo` class, could we merge both these classes in one?

Reply:
I would avoid doing so.  This class represents the low-level DER encoded structure and aligns exactly with the RFC specification.  `Rfc3161TimestampTokenInfo` is a more user-friendly wrapper that [handles endianness](https://github.com/NuGet/NuGet.Client/pull/2007/files#diff-047faaa2645e38cf9508ca16b5a642cbR116), [Win32 interop](https://github.com/NuGet/NuGet.Client/pull/2007/files#diff-047faaa2645e38cf9508ca16b5a642cbL114), etc.  This class replaces the old `Rfc3161TimestampTokenInfo.TstInfo`, which served the same purpose.